url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/Listener.java

```diff
@@ -0,0 +1,58 @@
+package ccproxy;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+
+import org.apache.log4j.Logger;
+
+public class Listener extends Thread {
+	static Logger log = Logger.getLogger(Listener.class);
+	ServerSocketChannel serverChan=null;
+	GadajAsterisk polaczenieAsterisk;
+	AgentConnectionList agenci;
+	DBConnection dbConn;
+	Parametry p;
+
+	public Listener(String port, Parametry param) {
+		super();
+		this.p = param;
+
+
+		try {
+			serverChan = ServerSocketChannel.open();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *serverChan*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).